### ISSUE TITLE
protobufjs 6.x support seemingly broken

### ISSUE THREAD

**Issue by pcarrier** [grpc/grpc#11484]
grpc 1.3.8 and 1.4.0-pre1 fail with protobufjs 6.8.0 with the following error in client logs:

> Error: Failed to parse server response

From interactive debugging sessions, `deserializeCls` in `protobuf_js_6_common.js` calls `toObject(...)` which does not exist on the value returned by `cls.decode(arg_buf)`. Removing `toObject(...)` addresses the issue.

Script to reproduce, should only need `npm install ssl-root-cas grpc protobufjs` in a recent Node:

```
const pbjs = require('protobufjs');
const grpc = require('grpc');
const cas = new Buffer(require('ssl-root-cas').join("\n"));

const service = grpc.loadObject(pbjs.Root.fromJSON({
    nested: {
        counter: {
            nested: {
                Counter: {
                    methods: {
                        Increment: {
                            requestType: 'CounterAmount', responseType: 'CountResponse'
                        }
                    }
                },
                CounterAmount: {
                    fields: {
                        name: {type: 'string', id: 1},
                        amount: {type: 'int64', id: 2}
                    }
                },
                CountResponse: {
                    fields: {
                        success: {type: 'bool', id: 1},
                        count: {type: 'int64', id: 2}
                    }
                }
            }
        }
    }
}));

const stub = new service.counter.Counter('counter-staging.apollodata.com:443', grpc.credentials.createSsl(cas));
api.increment({name: 'hello', amount: 1}, function(err, res) {
    if (err) {
        console.log("Error: " + err);
    }
    console.log("Result: " + JSON.stringify(res));
});
```

Before:

```
$ node repro.js
Error: Error: Failed to parse server response
Result: undefined
```

Change `node_modules/grpc/src/node/src/protobuf_js_6_common.js`:

```
-    return cls.decode(arg_buf).toObject(conversion_options);
+    return cls.decode(arg_buf);
```

After:

```
$ node repro.js
Result: {"success":true,"count":"12356"}
```

Credits to @omegablitz for most of the work here.
Cc @paralin @murgatroid99.

**Comment by murgatroid99** [comment]
I tested that code against Protobuf.js 6.7. Can you try downgrading and see if that fixes the problem?

**Comment by murgatroid99** [comment]
Update: this code works using Protobuf.js 6.7, and fails using Protobuf.js 6.8.

**Comment by murgatroid99** [comment]
This should be fixed in 1.3.9, and the fix will also be in the upcoming 1.4.1.
